### PR TITLE
Refactor the main menu into css`` and a flatter file structure

### PR DESCRIPTION
**Pull request by AWare**
## What does this change?
The main menu.

### CODE REVIEW

File: frontend/components/Header/Nav/MainMenu/Column.js [hunk 1]
```diff
@@ -0,0 +1,358 @@
...
+    toggleColumnLinks,
+    ariaControls,
+    isLastIndex,
+}: {
+    column: LinkType,
+    showColumnLinks: boolean,
+    toggleColumnLinks: () => void,
+    ariaControls: string,
+    isLastIndex: boolean,
+}) => (
```
Comment (GHaberis) [inline frontend/components/Header/Nav/MainMenu/Column.js]: According to the [flow docs](https://flow.org/en/docs/react/types/#toc-react-element) the way to type stateless components is: 
```js
const element: React.Element<'div'> = <div />;
```
However as mentioned in https://github.com/facebook/flow/issues/4130

> `import React from 'react'` doesn't exposes the React types anymore. So if you want to use `React.Element<*>`, you need to use `import * as React from 'react'`.

OR this should work:
```js
const element: React$Element<*> = <div />;
```

File: frontend/components/Header/Nav/MainMenu/Column.js [hunk 2]
```diff
@@ -0,0 +1,358 @@
...
+    width: 100%;
+    ${desktop}: {
+        display: list-item;
+    };
+`;
+
+const ColumnLink = ({ link, column }: { column: LinkType, link: LinkType }) => (
+    <li
+        className={cx(mainMenuLinkStyle, {
+            [hideDesktop]: link.mobileOnly || false,
```
[the remark this answers is not in the document]
Comment (GHaberis) [reply frontend/components/Header/Nav/MainMenu/Column.js]: `link.mobileOnly` could be `undefined`, so `!!link.mobileOnly` should do the job if we want `hideDesktop` to be passed as a boolean.

File: frontend/components/Header/Nav/MainMenu/Column.js [hunk 3]
```diff
@@ -0,0 +1,358 @@
...
+        transform: rotate(-135deg);
+    }
+`;
+const hideAfter = css`
+    :after {
+        display: none;
+    }
+`;
+
+const colapseColumnButton = css`
```
[the remark this answers is not in the document]
Comment (AWare) [reply frontend/components/Header/Nav/MainMenu/Column.js]: Oops.

File: frontend/components/Header/Nav/MainMenu/Column.js [hunk 4]
```diff
@@ -0,0 +1,358 @@
...
+const pillarStyles = pillar => css`
+    :hover {
+        color: ${pillars[pillar]};
+        text-decoration: underline;
+    }
+    :focus {
+        color: ${pillars[pillar]};
+        text-decoration: underline;
+    }
+`;
```
[the remark this answers is not in the document]
Comment (AWare) [reply frontend/components/Header/Nav/MainMenu/Column.js]: :+1:

File: frontend/components/Header/Nav/MainMenu/Column.js (hunk 1 above)
Comment (AWare) [reply frontend/components/Header/Nav/MainMenu/Column.js]: To me this is a benefit of typescript.

Comment (AWare) [reply frontend/components/Header/Nav/MainMenu/Column.js]: I think how we type React components like this probably merits wider discussion.

File: frontend/components/Header/Nav/MainMenu/Columns.js [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+    }
+    ${wide} {
+        width: 300px;
+    }
+`;
+
+const brandExtensionListItem = css`
+    marginright: 0;
+    margintop: -6px;
+    paddingbottom: 0;
```
[the remark this answers is not in the document]
Comment (AWare) [reply frontend/components/Header/Nav/MainMenu/Columns.js]: It really should.

File: frontend/components/Header/Nav/MainMenu/Column.js [hunk 6]
```diff
@@ -0,0 +1,355 @@
...
+        content: '';
+        display: block;
+        height: 1;
+        left: 50;
+        position: absolute;
+        width: 100%;
+    }
+`;
+
+const ColapseColumnButton = ({
```
Comment (GHaberis) [inline frontend/components/Header/Nav/MainMenu/Column.js]: Minor but this is a misspelling of "Collapse" as "Colapse"

File: frontend/components/Header/Nav/MainMenu/index.js [hunk 7]
```diff
@@ -8,67 +8,79 @@ import {
...
+        display: none;
+        position: absolute;
+        padding-bottom: 0;
+        padding-top: 0;
+        top: 100%;
+        left: 0;
+        right: 0;
+        width: 100%;
+        border-bottom: 1px solid #abc2c9;
+        @supports (width: 100vw) : {
```
Comment (GHaberis) [inline frontend/components/Header/Nav/MainMenu/index.js]: You don't need the ":" after `@supports (width: 100vw)`

**Pull request merged**